client/src/geoUtils.js

```diff
@@ -0,0 +1,53 @@
+import { forward, toPoint } from "mgrs"
+
+export function parseCoordinate(latLng) {
+  const value = parseFloat(latLng)
+  if (!value && value !== 0) {
+    return null
+  }
+
+  /*
+   * We use 5 decimal point (~110cm) precision because MGRS has
+   * a minimum of 1 meter precision.
+   * Please see;
+   * https://stackoverflow.com/a/16743805/1209097
+   * https://en.wikipedia.org/wiki/Military_Grid_Reference_System
+   *
+   * Additionally, coordinates must be truncated instead of rounding when changing
+   * precision level in order to aviod inconsistencies during (MGRS <--> Lat/Lon) conversion.
+   */
```

Comment:
This comment is no longer appropriate, you are now both rounding and truncating.

Reply:
Please update the comment and I think afterwards we're done.